packages/utilities/src/utilities/scrollIntoView/scrollIntoView.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright IBM Corp. 2016, 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * The viewport calculation for 12.5vh + 32px of the root margins.
+ *
+ * @private
+ */
+const _viewportMargin =
+  '-' +
+  (
+    Math.round(document.documentElement.clientHeight / (100 / 12.5)) + 32
+  ).toString() +
+  'px 0px';
+
+/**
+ * Intersection Observer options
+ *
+ * @private
+ */
+const _options = {
+  rootMargin: _viewportMargin,
+  threshold: 0,
+};
+
+/**
+ * Utility handles fade transition for selected elements.
+ *
+ * @example
+ * import '@carbon/ibmdotcom-styles/scss/internal/scroll-into-view/_scroll-into-view.scss';
+ * import { scrollIntoView } from '@carbon/ibmdotcom-utilities';
+ *
+ * As an example, the function can be called to target '.bx--content-block' as such:
+ *
+ * For default values of 400ms and continuous play:
+ * scrollIntoView('.bx--content-block')
+ *
+ * With 'one and done' option:
+ * scrollIntoView('.bx--content-block', false)
+ *
+ * For custom delay time, set within targeted class in the application's CSS code as such:
+ *
+ * .bx--content-block {
+ *   --#{$dds-prefix}--scroll-into-view-delay: 3s;
```

Comment:
Hm yeah, I originally wrote it like this, but since we're trying to move away from inline styling, I moved forward with the other approach by having the adopters set the delay within their own classes. I think both approaches have their pros and cons, but I do agree it'd be easier for the adopter to set it through the API. 

@asudoh I'm wondering if you have some thoughts about this?

Reply:
@IgnacioBecerra is right, we are avoiding defining style by JavaScript unless it's absolutely necessary. If we use inline style there is no way for application style to override it.

Reply:
@asudoh not sure I'm fully following that logic. What's a scenario that is "absolutely necessary"? The adopters would be the ones setting this, wouldn't they? This isn't stopping them from resetting it, does it? It seems odd to me they'd have to touch two separate files to get this working, and it uses two different sets of APIs. It also feels laborious to implement a proper delay that builds on top of itself using `SCSS`, and the logic that might be needed for the delay might be built out separately or duplicated. I don't know... Keep doing it if y'all are sure this is the right direction, but this doesn't jive in my brain.

Reply:
@photodow Application may want to define the rule of scrolling delay enforced to everywhere in the application, or everywhere in a specific portion. However, if a JavaScript code defines inline style, such inline style wins all the time and thus there is no way (that is natural in CSS world) for setting it application wide.

Reply:
@asudoh I'm not bought into the user experience you're creating here at the moment. I also don't know if we want other parts of the CSS overriding this value. I personally feel like this would be a scenario where inline would make sense, but for the sake of progress I'll step aside, and we can see where this takes us. Thanks as always @asudoh!

Reply:
Thanks @photodow for your thoughts! Another option is using an enum (that corresponds to a set of CSS classes) rather than a direct number.